src/device_properties.hpp

```diff
@@ -56,6 +56,17 @@ struct cvk_device_properties {
         return false;
     }
 
+    struct ClFormatSetCompare {
+        int operator()(const cl_image_format& lhs,
+                       const cl_image_format& rhs) const {
+            return lhs.image_channel_order > rhs.image_channel_order ||
+                   (lhs.image_channel_order == rhs.image_channel_order &&
+                    lhs.image_channel_data_type > rhs.image_channel_data_type);
+        }
+    };
+    using image_format_set = std::set<cl_image_format, ClFormatSetCompare>;
```

Comment:
Does the set need to be ordered?
- If not, can you reuse `ClFormatMapEqual`? (Maybe after a rename to remove `Map`)
- If yes, I suggest moving the functor to `image_format.hpp` with a rename to remove `Set`.

Reply:
The set does not need to be ordered, but reusing `clFromatMapEqual` leads to crashes on ChromeOS, so I'd prefer to stick with what is in this PR.
I can move the type to `image_format.hpp` but it feels weird to me as it creates a dependency of `device_properties.hpp` on `image_format.hpp` that is not really needed.

Reply:
Hmm, this sounds suspicious :). The dependency might not be too bad. I'll take a stab at this.